ouroboros-network/src/Ouroboros/Network/Pipe.hs

```diff
@@ -47,7 +46,10 @@ writePipe :: (Mx.ProtocolEnum ptcl, Ord ptcl, Enum ptcl, Bounded ptcl)
           -> IO (Time IO)
 writePipe ctx sdu = do
     ts <- getMonotonicTime
-    let sdu' = sdu { Mx.msTimestamp = Mx.RemoteClockModel $ fromIntegral $ ts .&. 0xffffffff }
+    let ts32 :: Word32
+               -- grab the low 32bits of the timestamp in microseconds
+        ts32 = fromIntegral (durationMicroseconds (ts `diffTime` zeroTime))
+    let sdu' = sdu { Mx.msTimestamp = Mx.RemoteClockModel ts32 }
```

Comment:
Why note move this to a "lower 32bit" operation in `MonadTimer.hs`? This will be needed for Pipes, Sockets and Queues.

Reply:
Done, utils in `O.N.Time`